src/components_new/DictionaryModal.tsx

```
@@ -0,0 +1,314 @@
+  onClose,
+  isPrivate,
+  session,
+  createdById,
+}) => {
+  const [newFrom, setNewFrom] = useState("");
+  const [newTo, setNewTo] = useState("");
+  const [editingId, setEditingId] = useState<string | null>(null);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const { data: dictionary = [], isLoading: isLoadingDictionary } = !isPrivate
+    ? useQuery(api.dictionary.getPublicDictionary, {
```

ESLint

[error] React Hook "useQuery" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)